Source/WTF/wtf/StdLibExtras.h

```diff
@@ -691,9 +691,10 @@ std::span<uint8_t, Extent == std::dynamic_extent ? std::dynamic_extent: Extent *
     return std::span<uint8_t, Extent == std::dynamic_extent ? std::dynamic_extent: Extent * sizeof(T)> { reinterpret_cast<uint8_t*>(span.data()), span.size_bytes() };
 }
 
-template<typename T, std::size_t Extent1, std::size_t Extent2>
-bool equalSpans(std::span<T, Extent1> a, std::span<T, Extent2> b)
+template<typename T, std::size_t TExtent, typename U, std::size_t UExtent>
+bool equalSpans(std::span<T, TExtent> a, std::span<U, UExtent> b)
 {
+    static_assert(sizeof(T) == sizeof(U));
```

Comment:
We should probably be static asserting or using concepts to make sure we get a compile error if you try to call this when `T` or `U` have non-trivial `==` implementations?

Reply:
I am not sure how to check for non-trivial implementations. Maybe I could static_assert for is_standard_layout?
https://en.cppreference.com/w/cpp/types/is_standard_layout

Reply:
The documentation for `std::memcmp` says it is typically meaningful for only trivially-copyable objects that have no padding. So I could check for `std::is_trivially_copy_constructible`.

Reply:
If there is any padding then `memcmp` won’t work. We do this kind of check in `HashTraits`, not by checking what is illegal, but by only allowing it on certain things that are known to be legal.

Wording on `cppreference.com` documentation for `memcmp` calls this “trivially-copyable objects that have no padding”, and while that’s a pretty indirect way to define it, I think that’s what we are trying to check. I’ll try to research this a tiny bit more.

Reply:
`std:: is_trivially_copyable_v` does exist, but not sure how to tackle the padding topic

Reply:
https://en.cppreference.com/w/cpp/types/has_unique_object_representations seems potentially interesting too.

Reply:
Maybe for now we can do a too-conservative padding test by just checking `std::is_scalar_v`?

Reply:
Oh, `has_unique_object_representations` is probably great. Honestly, we don’t have to get this 100% right, I just don’t want someone to write future bad code by accident by using this on something like `WTF::String`.